src/renderers/webgl/annotations/RectangleAnnotationRenderer.ts

```diff
@@ -0,0 +1,243 @@
+    this.resize(annotation.width, annotation.height).moveTo(annotation.x, annotation.y)
+    this.annotation = annotation
+  }
+
+  update(annotation: RectangleAnnotation) {
+    this.annotation = annotation
+
+    this.fill.update(annotation.style.color, annotation.style.opacity)
+    this.strokes?.update(annotation.style.stroke)
+
+    if (annotation.content && this.text) {
+      this.text.update(annotation.content, annotation.style.text)
+    } else if (!annotation.content && this.text) {
+      this.managers.text.delete(this.text)
+      this.text = undefined
+    }
+
+    const x = annotation.x
+    const y = annotation.y
+    const width = annotation.width
+    const height = annotation.height
+    const xChanged = x !== this.x
+    const yChanged = y !== this.y
+    const widthChanged = width !== this.width
+    const heightChanged = height !== this.height
+
+    if (xChanged || yChanged || widthChanged || heightChanged) {
+      // TODO -> enable in renderer options
+      if (xChanged) {
+        this.interpolateX = interpolate(this.x, x, DEFAULT_ANIMATE_RESIZE)
+      }
+      if (yChanged) {
+        this.interpolateY = interpolate(this.y, y, DEFAULT_ANIMATE_RESIZE)
+      }
+      if (widthChanged) {
+        this.interpolateWidth = interpolate(this.width, width, DEFAULT_ANIMATE_RESIZE)
+      }
+      if (heightChanged) {
+        this.interpolateHeight = interpolate(this.height, height, DEFAULT_ANIMATE_RESIZE)
+      }
+    } else {
+      this.resize(width, height).moveTo(x, y)
+      this.interpolateX = undefined
+      this.interpolateY = undefined
+      this.interpolateWidth = undefined
+      this.interpolateHeight = undefined
+    }
+
+    return this
+  }
+
+  render(dt: number) {
+    if (this.interpolateX || this.interpolateY) {
+      let x = this.x
+      let y = this.y
+
+      if (this.interpolateX) {
+        const { value, done } = this.interpolateX(dt)
+        x = value
+
+        if (done) {
+          this.interpolateX = undefined
+        }
+      }
+
+      if (this.interpolateY) {
+        const { value, done } = this.interpolateY(dt)
+        y = value
+
+        if (done) {
+          this.interpolateY = undefined
+        }
+      }
+
+      this.moveTo(x, y)
+    }
+
+    if (this.interpolateWidth || this.interpolateHeight) {
+      let width = this.width
+      let height = this.height
+
+      if (this.interpolateWidth) {
+        const { value, done } = this.interpolateWidth(dt)
+        width = value
+
+        if (done) {
+          this.interpolateWidth = undefined
+        }
+      }
+
+      if (this.interpolateHeight) {
+        const { value, done } = this.interpolateHeight(dt)
+        height = value
+
+        if (done) {
+          this.interpolateHeight = undefined
+        }
+      }
+
+      this.resize(width, height)
+    }
+
+    const isVisible = this.visible()
+
+    if (isVisible && this.annotation.content && !this.text) {
+      this.text = new Text(
```

Comment:
ditto, feels like initializing (and tearing down) the Text object here, and the RectangleStrokes object below, should happen in the update method. and only mounting/unmounting/moving/resizing should happen at render time.